boot/seal_test.go

```diff
@@ -235,10 +244,16 @@ func (s *sealSuite) TestSealKeyToModeenv(c *C) {
 		defer restore()
 
 		err = boot.SealKeyToModeenv(myKey, myKey2, model, modeenv)
-		if tc.sealErr != nil {
-			c.Assert(sealKeysCalls, Equals, 1)
+		c.Assert(provisionCalls, Equals, 1)
+		if tc.provisionErr != nil {
+			c.Assert(sealKeysCalls, Equals, 0)
+			continue
```

Comment:
This `continue` seems wrong at first sight, because if the test data are meant to trigger a provision error, then we should check that the `err` we received actually matches the expected one (this is done at line 261 below -- and there is another `continue` there, which seems appropriate).